tests/instruments/agilent/test_agilent33500.py

```diff
@@ -68,9 +67,9 @@ def test_frequency(frequency):
             ("SOUR1:FREQ?", frequency),
             ("SOUR2:FREQ?", frequency),
             ("FREQ?", frequency),
-            (f"SOUR1:FREQ {frequency}", None),
-            (f"SOUR2:FREQ {frequency}", None),
-            (f"FREQ {frequency}", None),
+            (f"SOUR1:FREQ {'{:.6f}'.format(frequency)}", None),
+            (f"SOUR2:FREQ {'{:.6f}'.format(frequency)}", None),
+            (f"FREQ {'{:.6f}'.format(frequency)}", None),
```

Comment:
Why do you nest two format strings into each other?
The following should be sufficient as it does the same:
```suggestion
            (f"SOUR1:FREQ {frequency:.6f}", None),
            (f"SOUR2:FREQ {frequency:.6f}", None),
            (f"FREQ {frequency:.6f}", None),
```

In curly braces you can write the variable name (or even an expression) directly.

Reply:
Changed as requested in https://github.com/pymeasure/pymeasure/pull/763/commits/bab560446383b62f8ab8a32e97eea8ba9d8ea071